### DISCUSSION TITLE
chapter_paperbook/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#1113] [Announcements]
# chapter_paperbook/

动画图解、一键运行的数据结构与算法教程

https://www.hello-algo.com/chapter_paperbook/

**Comment by ningblue** [comment]
大大真棒~

**Reply by harryer10** [reply]
支持

**Reply by ningblue** [reply]
也已经购买支持~

**Reply by RayLeeBoy** [reply]
怎么评价这本书呢, 这么说吧: 这是我第一本从头到尾仔细读完并且默写了书中大量代码实现的一本书, 完结撒花, 感谢k神及所有参与者的辛苦付出. 👍 🎉❤️

**Comment by SKRaua** [comment]
❤️❤️❤️

**Comment by weiyuexin** [comment]
必须支持

**Comment by Kukdo** [comment]
速速购入！

**Comment by 2sinx** [comment]
很6

**Comment by Harbour-z** [comment]
大爱，已支持

**Comment by ferryboy001** [comment]
这是我真正意义上购买的第一本算法书。

首先，我完全看不见，所以购买这本书对于我来说几乎没有任何实用价值。
为什么我选择购买这本书？
第一个原因是想要支持作者。因为真的想不到有什么更好的方式。作为我见到的第一本国内开源算法书籍，它的在线阅读的可访问性对于使用屏幕阅读器阅读的我来说实在是太惊喜了。
公式和图片可以说是视障者阅读电子书过程中的两大障碍。在这本书中，涉及到的所有数学公式，屏幕阅读器都可获取其 MathML 语义，并按照一定的规则进行朗读。一旦作者选择渲染为 svg 图像，我将无法阅读。
其次，网站对网页元素的合理使用，标题级别的合理安排，都提高了视障者浏览这本书的效率。屏幕阅读器提供了按各级标题浏览网页的方法，例如，按下大键盘的相应数字键就可以跳转到网页中的相应级别的标题。按下 H 可以按顺序跳转到下一个网页标题等等。
视障者阅读书籍并不能一目十行，即使屏幕阅读器语音的语速很快，也是线性的，使用标题和根据不同元素进行导航，可以从整体上了解文章的组织结构。
书籍中用于帮助读者理解知识点的图片内容，我无法阅读，这就需要我使用其他方法来试图了解。例如，我会拷贝图片到 Copilot 中询问，但也不能保证百分百准确，只能作为简单的参考。
所以，在阅读书籍的过程中，我会经常因为某个点不理解而卡住。
幸好这里有评论区，我可以随时询问。但很多时候，我不会选择第一时间询问，我总是希望能自己来搞懂这些。
稍微有点跑题了。第二个我购买这本书的原因是，想让它作为我书桌上的一部分，我想每天都能摸到它。我现在读大四，就读于针灸推拿学专业，因为是医学专业，所以是五年制。但我不喜欢我的专业，由于普通高考对于我来说存在太大的风险，例如我的成绩，就读什么大学的计算机相关专业，会不会被拒绝，我需要费多少口舌去交涉等等。最终我选择了通过盲校的单招考试进入本科。
可想而知，我的成绩一直不好，几乎从大学开始我就没有从学习中获得过什么快乐。
毕业和不确定的未来一直在我脑中盘旋，但我不知道该怎么做。
我经常在开学的时候看着学校计算机专业的课表，羡慕不已。但又强行自我安慰，国内的计算机专业并不值得我期待什么。
学习算法，让我可以了解到，如何高效解决问题。从指数级别的复杂度到线性级别的复杂度，是很奇妙的事情。如果阅读这条评论的你也在学习这本书，不论你为了什么目的，都希望你可以学到更多，加油！
最后，再次感谢作者，感谢所有回答过或将会回答我问题的朋友们！

**Reply by krahets** [reply]
谢谢你！你的评论为本书赋予了特别的意义。在设计网页版时（数学公式、网页元素、标题级别等），我致力于使显示效果简洁且优雅，却未曾深思你提到的“视障者浏览效率”问题。但现在看来，这竟是“无心插柳柳成荫”的意外收获。

虽然这本书富含“动画图解”，但目前缺乏能够有效将图片内容转化为文字描述的 AI 模型，这应该会给你的阅读带来不便。因此，我想分享两个计划：

- 未来若有好用的图片转文字的 AI 模型出现，我将努力将其整合进本书，以期为你们提供更完整的阅读体验。
- 今后我将致力于优化文字与图片的内容对应性，尽量保证即使只通过文字也能充分理解书中的内容。

仅从较短的评论中，我已经能感受到你的勇气与坚韧。在面临很多困难的情况下，你仍然能够在学习计算机的道路上不断前行，这真的很棒！愿功夫不负有心人，加油！

书中每一章的首页都有“花语”，我也写了一份送给你：

```text
我梦见微光，听见花开的声响，
在心灵深处，希望的花朵绽放。
未来如迷雾，我勇敢地向前望，
一步一脚印，迈向未知的远方。
```

**Reply by harry10086** [reply]
我不是学算法的，也不是学计算机的，只为你加油！

**Reply by MeowLiu** [reply]
读完你的评论感觉心里堵堵的，总之就是很感谢作者大大！同时，你真的很棒！

**Reply by brycenhe** [reply]
加油！

**Reply by litooorin** [reply]
加油！

**Comment by CryusXin** [comment]
今天偶然间看到同事买了这本书，当时还开玩笑说，等他看完借我看看，然后同事告诉我说有电子版的，于是我就打开了电子版的看了看，其实一开始我并没有什么太大的感觉，甚至觉得和其他的书一样，直到我看到讲递归的时候，我都还觉得递归好绕，但是看到讲斐波那契数列那个时候，一个f（n）就把我带回到高中数学函数了，突然领悟了什么是递归了，才发现这本书真的不简单，然后我就打开了京东，虽然我很不想承认我不喜欢看纸质书，但是还是为作者的开源精神所感悟，决定支持作者一次。

**Comment by Avengekkk** [comment]
已买，支持一下作者，希望什么时候可以出版一下C＋＋版本的，毕竟算法这东西还是C＋＋用着带劲。

**Reply by krahets** [reply]
谢谢支持～这个暂时还没有考虑，应该得从长计议了，先看看 Python 纸质书的表现

C++ 我也会一点，对比两者，我只想说：人生苦短，我选 Python

**Reply by litooorin** [reply]
你咋不说树剖呢，你咋不说AC自动机呢。并查集红黑树dijkstra应该不算入门吧

**Comment by KnewHow** [comment]
真希望高校可以用这本书当教材。

**Comment by xiellggl** [comment]
不知道是我的成长还是这书太好，我觉得很棒

**Comment by wangcuisun** [comment]
已经在京东买了，昨天刚到，拿到手很厚实的一本书，全彩印刷是真的棒，支持一下作者，看书不方便的时候就看网页版

**Comment by frankliuao** [comment]
这哥们是真牛逼。我一般不轻易公开道谢和表示我的崇拜，这哥们我可以，心服口服。行动力和知识都杠杠的。

**Reply by wcy168** [reply]
非常用心，书的质量非常的高，确实佩服。

**Comment by brycenhe** [comment]
支持一波！下单完成~

**Comment by c-z-zhao** [comment]
通俗易懂。半路出家，写了两年代码，很多东西能用，但是不理解为什么，读完本书通畅了。

**Comment by ifree-zm** [comment]
希望什么时候可以出一版java的

**Comment by Hwenray** [comment]
已购买支持，感谢~~

**Comment by TDSSSzero** [comment]
谢谢大佬们的耐心建设，已下单支持！

**Comment by Gmcc-sud0g** [comment]
说实话,看这书确实很形象,比一些古老生硬的好太多了,力推,已入手

**Comment by runshuui** [comment]
如果有c语言版就好了，对于大一学生真的很重要🙏🙏🙏

**Reply by krahets** [reply]
C 是工作，Python 是生活，可以一起学着 🤣

**Reply by wcmiracle** [reply]
其实学算法，语言是最无关紧要的，啥语言都行。

**Reply by tang9ian** [reply]
无所谓哪种语言，计算机根基都差不多

**Comment by chaofengdev** [comment]
难得遇到这么精美的书，作者真的花了很多时间精力来写这本书。可惜我已经过了看这个入门书的阶段了，要是当年有这本书，我该少走多少弯路啊。

**Comment by jungleford** [comment]
看完了。很精彩的一本基础入门读物。

如果作者还有时间和精力补充和完善的话，不妨再新增一章，专门介绍几个经典数据结构和算法，比如本书尚未讲到的：
* 并查集
* 红黑树（作为《树》一章AVL的扩展）
* Dijkstra算法，最大连通子图算法（加权图、有向图、DAG都可以作为《图》一章的扩展）

此外还可以再从科普视角给初学者简略介绍一点前沿知识，比如NLP或其他AI领域相关的算法知识。

**Reply by krahets** [reply]
谢谢建议！我们在 Issue #1086 中记录大家想看的新内容，欢迎各位反馈～

 > 此外还可以再从科普视角给初学者简略介绍一点前沿知识，比如NLP或其他AI领域相关的算法知识。

这是一个有趣的话题，值得思考一下该如何介绍

**Comment by uu1022** [comment]
666

**Comment by vipspecial** [comment]
很nice

**Comment by Brown2001** [comment]
已购买支持~~

**Comment by Zh1fen** [comment]
很棒的书

**Comment by qiany-sui** [comment]
已购买支持！

**Comment by zhouzhonghao** [comment]
已购买支持

**Comment by wyh136** [comment]
一本书传三代 算法永不过时  :p

**Reply by zmou** [reply]
一书传三代，人走书还在

**Reply by CCCsuper-minecraft** [reply]
666

**Comment by Jefy-Wang** [comment]
666

**Comment by CCCsuper-minecraft** [comment]
6666

**Comment by CCCsuper-minecraft** [comment]
资瓷

**Comment by xlzhou2012** [comment]
666

**Comment by One-M** [comment]
nice～

**Comment by jfhx** [comment]
做的太好了吧，这书

**Comment by AA34649** [comment]
nb

**Comment by 261237407** [comment]
6

**Comment by LucasLuQWQ** [comment]
nb
